examples/python/REACH_stream.py

```diff
@@ -6,90 +6,106 @@
 
-# path
+#--------SETTINGS--------#
 basedir = pathlib.Path('~/pyird/data/20211110_REACH/').expanduser()
 
-### For REACH ###
 inst = 'REACH'
+band = 'h' #'h' or 'y'
+mmf = 'mmf2' #'mmf1' (comb fiber) or 'mmf2' (star fiber)
+skipLFC = False #if False, uncertainties are output. mmf1 of y band must be reduced first.
 
-### FOR CALIBRATION ###
-# aperture extraction
+
+#--------FOR CALIBRATION--------#
+## FLAT_STAR
+# please change directry names and fits ids
 datadir = basedir/'flat/'
-anadir = basedir/'flat/'
-flat=irdstream.Stream2D("flat",datadir,anadir,inst=inst)
-flat.fitsid=list(range(53235,53334,2)) #no light in the speckle fiber
-################################
-### SELECT H band or YJ band ###
-################################
-flat.band='h' #'h' or 'y'
-print(flat.band,' band')
-if flat.band=='h':
-    flat.fitsid_increment() # when you use H-band
-    trace_smf=flat.aptrace(cutrow = 800,nap=42) #TraceAperture instance
-elif flat.band=='y':
-    trace_smf=flat.aptrace(cutrow = 1000,nap=102) #TraceAperture instance
+anadir = basedir/'reduc/'
+flat_star=irdstream.Stream2D("flat",datadir,anadir,inst=inst)
+flat_star.fitsid=list(range(53235,53334,2))
+# aperture extraction
+flat_star.band=band
+print(flat_star.band,' band')
+if band=='h' and flat_star.fitsid[0]%2==0:
+    flat_star.fitsid_increment() 
+    trace_smf=flat_star.aptrace(cutrow = 800,nap=42) 
+elif band=='y':
+    trace_smf=flat_star.aptrace(cutrow = 600,nap=102) 
 trace_mask = trace_smf.mask()
 
+# apeture mask plot
 import matplotlib.pyplot as plt
-plt.imshow(trace_smf.mask()) #apeture mask plot
+plt.imshow(trace_smf.mask()) 
 plt.show()
 
-# hotpixel mask: See pyird/io/read_hotpix.py for reading fixed mask (Optional)
+## HOTPIXEL MASK: 
+# See pyird/io/read_hotpix.py for reading fixed mask (Optional)
+## DARK
 datadir = basedir/'dark/'
 anadir = basedir/'dark/'
 dark = irdstream.Stream2D('dark', datadir, anadir,fitsid=[47269],inst=inst)
-if flat.band=='h':
-    dark.fitsid_increment() # when you use H-band
+if band=='h' and dark.fitsid[0]%2==0:
+    dark.fitsid_increment()
 median_image = dark.immedian()
 for data in dark.rawpath:
     im = pyf.open(str(data))[0].data
 im_subbias = bias_subtract_image(im)
 hotpix_mask = identify_hotpix_sigclip(im_subbias)
 
-###########################
-### SELECT mmf2 or mmf1 ###
-###########################
-trace_smf.mmf2() #mmf2 (star fiber)
-#trace_smf.mmf1() #mmf1 (comb fiber)
+# reduce mmf1 or mmf2
+if mmf=='mmf2':
+    trace_smf.mmf2() #mmf2 (star fiber)
+elif mmf=='mmf1':
+    trace_smf.mmf1() #mmf1 (comb fiber)
 
-# load ThAr raw image
+## THAR
 datadir = basedir/'thar'
-anadir = basedir/'thar'
-if flat.band=='h':
+anadir = basedir/'reduc'
+if band=='h':
     rawtag='IRDAD000'
-elif flat.band=='y':
+    ignore_orders=None
+elif band=='y':
     rawtag='IRDBD000'
+    ignore_orders = list(np.arange(1,32))+list([50,51])
+thar=irdstream.Stream2D("thar",datadir,anadir,fitsid=list(range(53347,53410,2)),inst=inst,rawtag=rawtag)#
 
```

Comment:
thar.band = band?

Reply:
Thank you for pointing this out!
It looks `rawtag` was mistakenly set for this data set, so the band had been set incorrectly. I fixed this by commenting out `rawtag` in Stream2D.
(We need to set `rawtag` for data starting from `IRDAD` or `IRDBD`, but do not have to data starting from `IRDA`.)